src/get-started/java-binding.md

```diff
@@ -0,0 +1,382 @@
+`exonum-java-binding-proofs`, are shipped together with the Exonum App, so when
+defining them in [pom.xml][pom.xml] you should use scope [`provided`](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope).
+
+### Service Development
+
+The service abstraction serves to expand the framework and implement the
+business logic of the application. The service defines the schema of the stored
+data that constitute the service state; transaction processing rules that can
+make changes to the stored data; and the external API that allows to interact
+with the service from outside of the system. See more information on the
+software model of the service in the [corresponding section][Exonum-services].
+
+In Java, the abstraction of the service is represented by
+`com.exonum.binding.service.Service` interface. Implementations can use the
+abstract class `com.exonum.binding.service.AbstractService`.
+
+### Service Schema Description
+
+Exonum provides a lot of stored named collections for the service data. The main
+collection types are sets, lists and maps. Data organization inside the
+collections is arranged in two ways - ordinary collections and [Merkelized
+collections](https://en.wikipedia.org/wiki/Merkle_tree); the latter allow to
+provide cryptographic evidence of the authenticity of the data to the clients of
+the system (for example, that an element is stored in the collection under a
+certain key). The [blockchain state](../glossary.md#blockchain-state) is
+influenced only by the Merkelized collections.
+
+For the detailed description of all Exonum collection types see the relevant
+[documentation](../architecture/storage/#table-types). In Java,
+implementations of the collections are located in
+`com.exonum.binding.storage.indices` package. This package documentation
+describes the peculiarities of their use.
+
+!!! note
+    `SparseListIndex` is not yet supported in Java. Let us know if it may be
+    useful for you!
+
+Collections work with a database view - either `Snapshot`, which is read-only;
+or `Fork`, which is mutable and allows performing modifying operations. The
+database representation is provided by the framework: `Snapshot` can be
+requested at any time, while `Fork` - only when the transaction is executed. The
+lifetime of these objects is limited by the scope of the method to which they
+are transferred.
+
+Exonum stores arrays of bytes in the storage collections. Therefore, the user
+must implement serialization of values ​​stored in the collection. Java Binding
+provides *serializers* for standard and some commonly used types, see
+`com.exonum.binding.storage.serialization.StandardSerializers`.
+
+!!! note "Example of ProofMapIndex Creation"
+    ```java
+    void updateBalance(Fork fork) {
+      var name = "balanceById";
+      var balanceById = ProofMapIndexProxy.newInstance(name, fork, StandardSerializers.hash(),
```

Comment:
Nit: It could make sense to make this line shorter for the sake of mobile device friendliness.